### PR TITLE
feat: read, update and delete travel

### PR DESCRIPTION
**Pull request by Neerfix**
#54

### CODE REVIEW

File: src/Controller/TravelController.php [hunk 1]
```diff
@@ -51,6 +52,141 @@ public function index(): Response
...
+        if (null === $travel) {
+            return $this->buildErrorResponse(Response::HTTP_NOT_FOUND, 'not.found', 'Le voyage demandé n\'est pas trouvé');
+        }
+
+        $roles = $loggedUser->getRoles();
+
+        // No logged used
+        if (in_array(User::ROLE_ADMIN, $roles, true) || $travel->getUserId() !== $loggedUser->getId()) {
+            return $this->buildErrorResponse(Response::HTTP_FORBIDDEN, 'auth.unauthorized', 'Vous n\'êtes pas autorisé à effectuer cette action');
+        }
```
Comment (Hikari-rom) [inline src/Controller/TravelController.php:156]:
```suggestion
        if (!in_array(User::ROLE_ADMIN, $roles, true) || $travel->getUserId() !== $loggedUser->getId()) {
            return $this->buildErrorResponse(Response::HTTP_FORBIDDEN, 'auth.unauthorized', 'Vous n\'êtes pas autorisé à effectuer cette action');
        }
```

File: src/Controller/TravelController.php [hunk 2]
```diff
@@ -51,6 +52,141 @@ public function index(): Response
...
+            $loggedUser
+        );
+
+        $travelNormalizer = $this->normalizer->normalize($travelDeleted, null, ['groups' => ['travel:read', 'travel:nested']]);
+
+        return $this->buildSuccessResponse(Response::HTTP_OK, $travelNormalizer, $loggedUser);
+    }
+
+    /**
+     * @Route("/travel/{uuid}", name="update_travel", methods={ "PUT" })
```
Comment (Neerfix) [inline src/Controller/TravelController.php:129]:
```suggestion
     * @Route("/travels/{uuid}", name="update_travel", methods={ "PUT" })
```

File: src/Controller/TravelController.php [hunk 3]
```diff
@@ -51,6 +52,141 @@ public function index(): Response
...
+            return $this->buildErrorResponse(Response::HTTP_NOT_FOUND, 'not.found', 'Le voyage demandé n\'est pas trouvé');
+        }
+
+        $roles = $loggedUser->getRoles();
+
+        // No logged used
+        if (in_array(User::ROLE_ADMIN, $roles, true) || $travel->getUserId() !== $loggedUser->getId()) {
+            return $this->buildErrorResponse(Response::HTTP_FORBIDDEN, 'auth.unauthorized', 'Vous n\'êtes pas autorisé à effectuer cette action');
+        }
+
```
Comment (Hikari-rom) [inline src/Controller/TravelController.php:83]:
```suggestion
        if (!in_array(User::ROLE_ADMIN, $roles, true) || $travel->getUserId() !== $loggedUser->getId()) {
            return $this->buildErrorResponse(Response::HTTP_FORBIDDEN, 'auth.unauthorized', 'Vous n\'êtes pas autorisé à effectuer cette action');
        }

```

**Review by Hikari-rom — CHANGES_REQUESTED** [review]

File: src/Controller/TravelController.php [hunk 4]
```diff
@@ -51,6 +52,141 @@ public function index(): Response
         return $this->buildSuccessResponse(Response::HTTP_OK, $travel, $loggedUser);
     }
 
+    /**
+     * @Route("/travel/{uuid}", name="get_travel", methods={ "GET" })
```
Comment (Hikari-rom) [inline src/Controller/TravelController.php:56]:
```suggestion
     * @Route("/travels/{uuid}", name="get_travel", methods={ "GET" })
```

File: src/Controller/TravelController.php [hunk 5]
```diff
@@ -51,6 +52,141 @@ public function index(): Response
...
+
+        $travelNormalizer = $this->normalizer->normalize($travel, null, ['groups' => ['travel:read', 'travel:nested']]);
+
+        return $this->buildSuccessResponse(Response::HTTP_OK, $travelNormalizer, $loggedUser);
+    }
+
+    /**
+     * @Route("/travel/{uuid}", name="delete_travel", methods={ "DELETE" })
+     *
+     * @throws Exception
```
Comment (Hikari-rom) [inline src/Controller/TravelController.php:92]:
```suggestion
     * @Route("/travels/{uuid}", name="delete_travel", methods={ "DELETE" })
     *
     * @throws Exception
```

**Review by Hikari-rom — CHANGES_REQUESTED** [review]
Just travels ;)

**Review by Hikari-rom — APPROVED** [review]
LGTM 🚀

**Pull request merged**